api/v1alpha1/policy_types.go

```diff
@@ -60,6 +65,35 @@ func (d *Policy) IsInitialized() bool {
 }
 
 func (d *Policy) PrepareInternalValues(context context.Context, object client.Object) error {
+	// Fast path escape if no "${..}" placeholder is detected
+	match, err := regexp.MatchString("\\${[^}]+}", d.Spec.Policy)
+	if err != nil || !match {
+		return nil
+	}
+
+	log := log.FromContext(context)
+
+	// Retrieves the list of auth engines to get their accessors
+	// Kinda duplicates logic found in VaultEngineObject.retrieveAccessor
+	vaultClient := context.Value("vaultClient").(*vault.Client)
+	secret, err := vaultClient.Logical().Read("sys/auth")
+	if err != nil {
```

Comment:
I understand why you mapped this to an error, but should we? This would mean that now everyone who is trying to create a policy must also have access to the sys/auth path. Maybe that is not desirable. What about just logging the problem and continuing with an error and of course without replacement.
Actually, even better can we check if there is anything to replace and if there is then we return an error?

Reply:
I'm not sure I fully understand the question...

> This would mean that now everyone who is trying to create a policy must also have access to the sys/auth path.

Actually, no: if there is no `${...}` placeholder in the policy rules (checked at line 69), then line 70 will exit without further processing and no error - no change compared to existing code.

In case there is at least one `${...}` placeholder but an error is returned by `sys/auth`, then the error is logged at line 80 but no error is returned (same behavior as existing code, no replacement done nor error raised).

> What about just logging the problem and continuing with an error and of course without replacement.

Yes, I believe that this is the behavior that is implemented, the only case when an error is raised is when there is at least one placeholder to replace but sys/auth returned nothing, without error.
Unless I've missed sthg, of course ;-)

> Actually, even better can we check if there is anything to replace and if there is then we return an error?

This check is done at the very beginning of the method, see above.

Reply:
@raffaelespazzoli any news or anything I can provide to help getting that through?
Thx